### ISSUE TITLE
CLI with default_config gets multiple values for `params_file`

### ISSUE THREAD

**Issue by arokem** [yeatmanlab/pyAFQ#353]
I edited the default config file only to point to the BIDS directory where the data is living (this is still the data from #261). I am getting:

```
(afq) ➜  pyAFQ git:(master) ✗ pyAFQ default_config.toml
Loading AFQ libraries...
INFO:pyAFQ:Your use of pyAFQ is being recorded using Google Analytics. For more information, please refer to the pyAFQ documentation: https://yeatmanlab.github.io/pyAFQ/usage.html#usage-tracking-with-google-analytics. To turn this off, use the `--notrack` flag when using the pyAFQ CLI
INFO:dipy.data.fetcher:Dataset is already in place. If you want to fetch it again please first remove the folder /Users/arokem/AFQ_data/templates 
INFO:dipy.data.fetcher:Dataset is already in place. If you want to fetch it again please first remove the folder /Users/arokem/AFQ_data/callosum_templates 
/Users/arokem/miniconda3/envs/afq/lib/python3.8/site-packages/bids/layout/models.py:98: FutureWarning: The 'extension' entity currently excludes the leading dot ('.'). As of version 0.14.0, it will include the leading dot. To suppress this warning and include the leading dot, use `bids.config.set_option('extension_initial_dot', True)`.
  warnings.warn("The 'extension' entity currently excludes the leading dot ('.'). "
INFO:AFQ.api:Saving /Users/arokem/AFQ_data/duanliangtao2/derivatives/afq/sub-01/ses-01/sub-01_ses-01_dwi_b0.nii.gz
INFO:AFQ.api:Saving /Users/arokem/AFQ_data/duanliangtao2/derivatives/afq/sub-01/ses-01/sub-01_ses-01_dwi_brain_mask.nii.gz
INFO:AFQ.api:Saving /Users/arokem/AFQ_data/duanliangtao2/derivatives/afq/sub-01/ses-01/sub-01_ses-01_dwi_model-DTI_diffmodel.nii.gz
INFO:AFQ.api:Saving /Users/arokem/AFQ_data/duanliangtao2/derivatives/afq/sub-01/ses-01/sub-01_ses-01_dwi_model-DTI_desc-DEC_FA.nii.gz
INFO:AFQ.api:Saving /Users/arokem/AFQ_data/duanliangtao2/derivatives/afq/sub-01/ses-01/sub-01_ses-01_dwi_model-DTI_PDD.nii.gz
INFO:AFQ.api:Saving /Users/arokem/AFQ_data/duanliangtao2/derivatives/afq/sub-01/ses-01/sub-01_ses-01_dwi_anisotropic_power_map.nii.gz
Optimizing level 2 [max iter: 10000]
Optimizing level 1 [max iter: 1000]
Optimizing level 0 [max iter: 100]
Optimizing level 2 [max iter: 10000]
Optimizing level 1 [max iter: 1000]
Optimizing level 0 [max iter: 100]
Optimizing level 2 [max iter: 10000]
Optimizing level 1 [max iter: 1000]
Optimizing level 0 [max iter: 100]
INFO:dipy.align.imwarp:Creating scale space from the moving image. Levels: 3. Sigma factor: 0.200000.
INFO:dipy.align.imwarp:Creating scale space from the static image. Levels: 3. Sigma factor: 0.200000.
INFO:dipy.align.imwarp:Optimizing level 2
INFO:dipy.align.imwarp:Optimizing level 1
INFO:dipy.align.imwarp:Optimizing level 0
INFO:AFQ.api:Saving /Users/arokem/AFQ_data/duanliangtao2/derivatives/afq/sub-01/ses-01/sub-01_ses-01_dwi_b0_in_MNI.nii.gz
INFO:AFQ.api:Saving /Users/arokem/AFQ_data/duanliangtao2/derivatives/afq/sub-01/ses-01/sub-01_ses-01_dwi_template_xform.nii.gz
Traceback (most recent call last):
  File "/Users/arokem/miniconda3/envs/afq/bin/pyAFQ", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/Users/arokem/source/pyAFQ/bin/pyAFQ", line 83, in <module>
    afb.parse_config_run_afq(
  File "/Users/arokem/source/pyAFQ/AFQ/utils/bin.py", line 235, in parse_config_run_afq
    myafq.export_all()
  File "/Users/arokem/source/pyAFQ/AFQ/api.py", line 1752, in export_all
    self.export_bundles()
  File "/Users/arokem/source/pyAFQ/AFQ/api.py", line 1685, in export_bundles
    self.data_frame.apply(self._export_bundles, axis=1)
  File "/Users/arokem/miniconda3/envs/afq/lib/python3.8/site-packages/pandas/core/frame.py", line 6878, in apply
    return op.get_result()
  File "/Users/arokem/miniconda3/envs/afq/lib/python3.8/site-packages/pandas/core/apply.py", line 186, in get_result
    return self.apply_standard()
  File "/Users/arokem/miniconda3/envs/afq/lib/python3.8/site-packages/pandas/core/apply.py", line 295, in apply_standard
    result = libreduction.compute_reduction(
  File "pandas/_libs/reduction.pyx", line 620, in pandas._libs.reduction.compute_reduction
  File "pandas/_libs/reduction.pyx", line 128, in pandas._libs.reduction.Reducer.get_result
  File "/Users/arokem/source/pyAFQ/AFQ/api.py", line 1201, in _export_bundles
    bundles_file = func(row)
  File "/Users/arokem/source/pyAFQ/AFQ/api.py", line 1018, in _clean_bundles
    bundles_file = self._segment(row)
  File "/Users/arokem/source/pyAFQ/AFQ/api.py", line 974, in _segment
    streamlines_file = self._streamlines(row)
  File "/Users/arokem/source/pyAFQ/AFQ/api.py", line 933, in _streamlines
    sft = aft.track(params_file, **tracking_params)
TypeError: track() got multiple values for argument 'params_file'
```

**Comment by 36000** [comment]
I see the problem. params_file is not an optional argument, so I need to ignore it. Only optional arguments should be in the config file.

**Comment by arokem** [comment]
If I comment that out, I also get the following, which seems related:

```
(afq) ➜  pyAFQ git:(master) ✗ pyAFQ default_config.toml
Loading AFQ libraries...
INFO:pyAFQ:Your use of pyAFQ is being recorded using Google Analytics. For more information, please refer to the pyAFQ documentation: https://yeatmanlab.github.io/pyAFQ/usage.html#usage-tracking-with-google-analytics. To turn this off, use the `--notrack` flag when using the pyAFQ CLI
INFO:dipy.data.fetcher:Dataset is already in place. If you want to fetch it again please first remove the folder /Users/arokem/AFQ_data/templates 
INFO:dipy.data.fetcher:Dataset is already in place. If you want to fetch it again please first remove the folder /Users/arokem/AFQ_data/callosum_templates 
/Users/arokem/miniconda3/envs/afq/lib/python3.8/site-packages/bids/layout/models.py:98: FutureWarning: The 'extension' entity currently excludes the leading dot ('.'). As of version 0.14.0, it will include the leading dot. To suppress this warning and include the leading dot, use `bids.config.set_option('extension_initial_dot', True)`.
  warnings.warn("The 'extension' entity currently excludes the leading dot ('.'). "
INFO:AFQ.tractography:Loading Image...
INFO:AFQ.tractography:Generating Seeds...
INFO:AFQ.tractography:Getting Directions...
100%|█████████████████████████████████████████| 100/100 [00:10<00:00, 16.21it/s]
INFO:AFQ.tractography:Tracking...
100%|██████████████████████████████████| 720896/720896 [39:58<00:00, 300.53it/s]
INFO:AFQ.api:Saving /Users/arokem/AFQ_data/duanliangtao2/derivatives/afq/sub-01/ses-01/sub-01_ses-01_dwi_space-RASMM_model-DTI_desc-det_tractography.trk
INFO:AFQ.Segmentation:Preparing Segmentation Parameters
INFO:AFQ.Segmentation:Preprocessing Streamlines
INFO:dipy.data.fetcher:Dataset is already in place. If you want to fetch it again please first remove the folder /Users/arokem/AFQ_data/aal_atlas 
INFO:AFQ.Segmentation:Assigning Streamlines to Bundles
INFO:AFQ.Segmentation:Finding Streamlines for ATR_R
INFO:AFQ.Segmentation:15818 streamlines exceed the probability threshold.
100%|███████████████████████████████████| 15818/15818 [00:01<00:00, 9531.04it/s]
INFO:AFQ.Segmentation:0 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for ATR_L
INFO:AFQ.Segmentation:16542 streamlines exceed the probability threshold.
100%|███████████████████████████████████| 16542/16542 [00:02<00:00, 6596.71it/s]
INFO:AFQ.Segmentation:85 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for CGC_R
INFO:AFQ.Segmentation:3473 streamlines exceed the probability threshold.
100%|█████████████████████████████████████| 3473/3473 [00:00<00:00, 5795.67it/s]
INFO:AFQ.Segmentation:0 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for CGC_L
INFO:AFQ.Segmentation:3991 streamlines exceed the probability threshold.
100%|█████████████████████████████████████| 3991/3991 [00:01<00:00, 2271.64it/s]
INFO:AFQ.Segmentation:5 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for CST_R
INFO:AFQ.Segmentation:10299 streamlines exceed the probability threshold.
100%|██████████████████████████████████| 10299/10299 [00:00<00:00, 13440.45it/s]
INFO:AFQ.Segmentation:62 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for CST_L
INFO:AFQ.Segmentation:10530 streamlines exceed the probability threshold.
100%|███████████████████████████████████| 10530/10530 [00:02<00:00, 4535.92it/s]
INFO:AFQ.Segmentation:297 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for IFO_R
INFO:AFQ.Segmentation:17424 streamlines exceed the probability threshold.
100%|███████████████████████████████████| 17424/17424 [00:09<00:00, 1786.32it/s]
INFO:AFQ.Segmentation:12 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for IFO_L
INFO:AFQ.Segmentation:16092 streamlines exceed the probability threshold.
100%|███████████████████████████████████| 16092/16092 [00:13<00:00, 1208.75it/s]
INFO:AFQ.Segmentation:38 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for ILF_R
INFO:AFQ.Segmentation:13483 streamlines exceed the probability threshold.
100%|████████████████████████████████████| 13483/13483 [00:25<00:00, 530.82it/s]
INFO:AFQ.Segmentation:777 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for ILF_L
INFO:AFQ.Segmentation:10429 streamlines exceed the probability threshold.
100%|████████████████████████████████████| 10429/10429 [00:23<00:00, 439.16it/s]
INFO:AFQ.Segmentation:250 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for SLF_R
INFO:AFQ.Segmentation:16234 streamlines exceed the probability threshold.
100%|█████████████████████████████████████| 16234/16234 [30:31<00:00,  8.86it/s]
INFO:AFQ.Segmentation:1426 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for SLF_L
INFO:AFQ.Segmentation:14408 streamlines exceed the probability threshold.
100%|████████████████████████████████████| 14408/14408 [00:27<00:00, 518.78it/s]
INFO:AFQ.Segmentation:1470 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for ARC_R
INFO:AFQ.Segmentation:10960 streamlines exceed the probability threshold.
100%|████████████████████████████████████| 10960/10960 [00:10<00:00, 997.56it/s]
INFO:AFQ.Segmentation:8 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for ARC_L
INFO:AFQ.Segmentation:10448 streamlines exceed the probability threshold.
100%|████████████████████████████████████| 10448/10448 [00:12<00:00, 822.63it/s]
INFO:AFQ.Segmentation:53 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for UNC_R
INFO:AFQ.Segmentation:6684 streamlines exceed the probability threshold.
100%|█████████████████████████████████████| 6684/6684 [00:02<00:00, 2566.88it/s]
INFO:AFQ.Segmentation:65 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for UNC_L
INFO:AFQ.Segmentation:7549 streamlines exceed the probability threshold.
100%|█████████████████████████████████████| 7549/7549 [00:03<00:00, 2317.03it/s]
INFO:AFQ.Segmentation:198 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for FA
INFO:AFQ.Segmentation:12368 streamlines exceed the probability threshold.
100%|████████████████████████████████████| 12368/12368 [00:21<00:00, 364.70it/s]
INFO:AFQ.Segmentation:104 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Finding Streamlines for FP
INFO:AFQ.Segmentation:8692 streamlines exceed the probability threshold.
100%|██████████████████████████████████████| 8692/8692 [00:12<00:00, 679.56it/s]
INFO:AFQ.Segmentation:4 streamlines selected with waypoint ROIs
INFO:AFQ.Segmentation:Re-orienting streamlines to consistent directions
INFO:AFQ.Segmentation:Processing ATR_R
INFO:AFQ.Segmentation:Processing ATR_L
INFO:AFQ.Segmentation:Filtering by endpoints
INFO:AFQ.Segmentation:Before filtering 81 streamlines
INFO:AFQ.Segmentation:After filtering 8 streamlines
INFO:AFQ.Segmentation:Processing CGC_R
INFO:AFQ.Segmentation:Processing CGC_L
INFO:AFQ.Segmentation:Filtering by endpoints
INFO:AFQ.Segmentation:Before filtering 5 streamlines
INFO:AFQ.Segmentation:After filtering 1 streamlines
INFO:AFQ.Segmentation:Processing CST_R
INFO:AFQ.Segmentation:Filtering by endpoints
INFO:AFQ.Segmentation:Before filtering 38 streamlines
INFO:AFQ.Segmentation:After filtering 34 streamlines
INFO:AFQ.Segmentation:Processing CST_L
INFO:AFQ.Segmentation:Filtering by endpoints
INFO:AFQ.Segmentation:Before filtering 297 streamlines
INFO:AFQ.Segmentation:After filtering 135 streamlines
INFO:AFQ.Segmentation:Processing IFO_R
INFO:AFQ.Segmentation:Filtering by endpoints
INFO:AFQ.Segmentation:Before filtering 4 streamlines
INFO:AFQ.Segmentation:After filtering 2 streamlines
INFO:AFQ.Segmentation:Processing IFO_L
INFO:AFQ.Segmentation:Filtering by endpoints
INFO:AFQ.Segmentation:Before filtering 19 streamlines
INFO:AFQ.Segmentation:After filtering 1 streamlines
INFO:AFQ.Segmentation:Processing ILF_R
INFO:AFQ.Segmentation:Filtering by endpoints
INFO:AFQ.Segmentation:Before filtering 729 streamlines
INFO:AFQ.Segmentation:After filtering 5 streamlines
INFO:AFQ.Segmentation:Processing ILF_L
INFO:AFQ.Segmentation:Filtering by endpoints
INFO:AFQ.Segmentation:Before filtering 166 streamlines
INFO:AFQ.Segmentation:After filtering 14 streamlines
INFO:AFQ.Segmentation:Processing SLF_R
INFO:AFQ.Segmentation:Filtering by endpoints
INFO:AFQ.Segmentation:Before filtering 1414 streamlines
INFO:AFQ.Segmentation:After filtering 50 streamlines
INFO:AFQ.Segmentation:Processing SLF_L
INFO:AFQ.Segmentation:Filtering by endpoints
INFO:AFQ.Segmentation:Before filtering 1446 streamlines
INFO:AFQ.Segmentation:After filtering 77 streamlines
INFO:AFQ.Segmentation:Processing ARC_R
INFO:AFQ.Segmentation:Filtering by endpoints
INFO:AFQ.Segmentation:Before filtering 6 streamlines
INFO:AFQ.Segmentation:Processing ARC_L
INFO:AFQ.Segmentation:Filtering by endpoints
INFO:AFQ.Segmentation:Before filtering 53 streamlines
INFO:AFQ.Segmentation:After filtering 34 streamlines
INFO:AFQ.Segmentation:Processing UNC_R
INFO:AFQ.Segmentation:Filtering by endpoints
INFO:AFQ.Segmentation:Before filtering 54 streamlines
INFO:AFQ.Segmentation:Processing UNC_L
INFO:AFQ.Segmentation:Filtering by endpoints
INFO:AFQ.Segmentation:Before filtering 194 streamlines
INFO:AFQ.Segmentation:After filtering 6 streamlines
INFO:AFQ.Segmentation:Processing FA
INFO:AFQ.Segmentation:Filtering by endpoints
INFO:AFQ.Segmentation:Before filtering 104 streamlines
INFO:AFQ.Segmentation:After filtering 12 streamlines
INFO:AFQ.Segmentation:Processing FP
INFO:AFQ.Segmentation:Filtering by endpoints
INFO:AFQ.Segmentation:Before filtering 4 streamlines
INFO:AFQ.api:Saving /Users/arokem/AFQ_data/duanliangtao2/derivatives/afq/sub-01/ses-01/sub-01_ses-01_dwi_space-RASMM_model-DTI_desc-det-AFQ_tractography.trk
Traceback (most recent call last):
  File "/Users/arokem/miniconda3/envs/afq/bin/pyAFQ", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/Users/arokem/source/pyAFQ/bin/pyAFQ", line 83, in <module>
    afb.parse_config_run_afq(
  File "/Users/arokem/source/pyAFQ/AFQ/utils/bin.py", line 235, in parse_config_run_afq
    myafq.export_all()
  File "/Users/arokem/source/pyAFQ/AFQ/api.py", line 1752, in export_all
    self.export_bundles()
  File "/Users/arokem/source/pyAFQ/AFQ/api.py", line 1685, in export_bundles
    self.data_frame.apply(self._export_bundles, axis=1)
  File "/Users/arokem/miniconda3/envs/afq/lib/python3.8/site-packages/pandas/core/frame.py", line 6878, in apply
    return op.get_result()
  File "/Users/arokem/miniconda3/envs/afq/lib/python3.8/site-packages/pandas/core/apply.py", line 186, in get_result
    return self.apply_standard()
  File "/Users/arokem/miniconda3/envs/afq/lib/python3.8/site-packages/pandas/core/apply.py", line 295, in apply_standard
    result = libreduction.compute_reduction(
  File "pandas/_libs/reduction.pyx", line 620, in pandas._libs.reduction.compute_reduction
  File "pandas/_libs/reduction.pyx", line 128, in pandas._libs.reduction.Reducer.get_result
  File "/Users/arokem/source/pyAFQ/AFQ/api.py", line 1201, in _export_bundles
    bundles_file = func(row)
  File "/Users/arokem/source/pyAFQ/AFQ/api.py", line 1036, in _clean_bundles
    this_tg = seg.clean_bundle(this_tg, **self.clean_params)
TypeError: clean_bundle() got multiple values for argument 'tg'

```

**Comment by arokem** [comment]
Maybe we need to run a test with the default config all the way through, using very few seeds (to make this not unbearably long)?

**Comment by 36000** [comment]
Yes, those should be the only two you need to comment. This PR will fix: https://github.com/yeatmanlab/pyAFQ/pull/354

**Comment by 36000** [comment]
I can add one. I could also mark it as long, so it doesn't always run, and then only run it on PRs that seem relevant.

**Comment by arokem** [comment]
Indeed. This seems resolved by #354
